server/src-lib/Hasura/GraphQL/Schema.hs

```diff
@@ -766,19 +773,19 @@ instance Monoid TyAgg where
 -- | A role-specific mapping from root field names to allowed operations.
 data RootFields
   = RootFields
-  { rootQueryFields    :: !(Map.HashMap G.Name (QueryCtx, ObjFldInfo))
-  , rootMutationFields :: !(Map.HashMap G.Name (MutationCtx, ObjFldInfo))
+  { rootQueryFields    :: ![(QueryCtx, ObjFldInfo)]
+  , rootMutationFields :: ![(MutationCtx, ObjFldInfo)]
```

Comment:
Related to the above comment: maybe we could keep these as `HashMap`s, and we could make the validation function produce them from the lists? A la the example at the end of my blog post.